dot/network/service.go

```diff
@@ -290,6 +320,45 @@ func (s *Service) collectNetworkMetrics() {
 	}
 }
 
+func (s *Service) getTotalStreams(inbound bool) (count int64) {
+	for _, conn := range s.host.h.Network().Conns() {
+		for _, stream := range conn.GetStreams() {
+			if (isInbound(stream) && inbound) || (!isInbound(stream) && !inbound) {
```

Comment:
I think we are computing isInbound twice for outbound streams. Change it to

```suggestion
			streamIsInbound := isInbound(stream)
			if (streamIsInbound && inbound) || (!streamIsInbound & !inbound) {
```

Unless isInbound is trivial/fast.

Reply:
updated